### PR TITLE
Fixed a count collect bug

### PR DESCRIPTION
**Pull request by hkernbach**
Backport of: https://github.com/arangodb/arangodb/pull/13553
Please see full details there.

### CODE REVIEW

**Comment by hkernbach** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13998/

**Comment by hkernbach** [comment]
image

File: tests/js/server/aql/aql-queries-optimizer-count.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      // create 100 (50+50) documents in total
+      for (let outer = 0; outer < 50; outer++) {
+        let myArray = [];
+        // create 100 entries for our inner document array
+        for (let inner = 0; inner < 10; inner++) {
+          // write numeric strings
+          myArray.push(JSON.stringify(inner));
+        }
+        collection.save({"someBoolean": true, "someArray": myArray});
+        collection.save({"someBoolean": false, "someArray": myArray});
```
Comment (goedderz) [inline tests/js/server/aql/aql-queries-optimizer-count.js:66]: Better save all documents in an array, and insert all at once. That saves time when running the tests.

File: tests/js/server/aql/aql-queries-optimizer-count.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      let resCountWithCollect = getQueryResults(countWithCollectQuery);
+
+      assertEqual(1, resCountWithLength.length);
+      assertEqual(1, resCountWithCollect.length);
+      assertEqual(resCountWithLength[0], 100);
+      assertEqual(resCountWithCollect[0], 100);
+
+      assertEqual(resCountWithCollect[0], resCountWithLength[0]);
+    }
+  };
```
Comment (goedderz) [inline tests/js/server/aql/aql-queries-optimizer-count.js:96]: There should be an assertion that the index is actually used by the `FOR` operation to make sure we test the correct thing. Or alternatively, add an [index hint](https://www.arangodb.com/docs/stable/aql/operations-for.html#indexhint) with `forceIndexHint: true` for the same effect.

**Review by goedderz — CHANGES_REQUESTED** [review]
A minor comment in the tests, otherwise LGTM.

**Comment by hkernbach** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/14001/

**Review by goedderz — APPROVED** [review]

**Comment by hkernbach** [comment]
jenkins blue.

**Pull request merged**